pandas/indexes/base.py

```diff
@@ -2886,6 +2888,9 @@ def join(self, other, how='left', level=None, return_indexers=False):
         elif how == 'outer':
             join_index = self.union(other)
 
+        if sort:
```

Comment:
@albertvillanova what I mean is that reading the code it is *very* confusing and not obvious how ``sort`` interacts with ``how``.

What I mean is I would explicity set ``sort=False`` (and maybe a comment) like this:

```
        if how == 'left':
            join_index = self
            # sort=True is ignored? if so, then set sort=False
       elif how == 'right':
            join_index = other
           # same
        elif how == 'inner':
            join_index = self.intersection(other)
        elif how == 'outer':
            join_index = self.union(other)
            # or could just sort=False
            assert sort is False, "outer join must explicty have sort=False"

        if sort:
            join_index = join_index.sort_values()
```